src/app.js

```diff
@@ -6,8 +6,225 @@ import _ from 'underscore';
+        handleValidationFailures(response.responseJSON["errors"]);
+      },
+    });
+  };
+  //////////////////////////
+  // reservation handler ///
+  //////////////////////////
+  const reserveTemplateShow = function reserveTemplateShow(trip){
+    const reservationElement = $('#reservation-form');
+    reservationElement.html('');
+    const generatedHTMLreserve = reserveTemplate(trip.attributes);
+    console.log(trip.attributes);
+    reservationElement.append(generatedHTMLreserve);
+  };
+
+  $('#reserve').show();
+  $('#reserve').on('click', function(event) {
+    reserveTemplateShow(trip);
+    $('#reserve').hide();
+    $('#add-reservation').on('submit', function(event) {
+      event.preventDefault();
+      console.log('on submit');
+      reserveTrip(event);
+    });
+
+  });
+};
+
+// pulled out fetchTripDetails function above
+const fetchTripDetails = function fetchTripDetails(event) {
+  $('#add-reservation').hide();
+  let tripId = $(this).attr('data-id');
+  let trip = tripList.get(tripId)
+  trip.fetch({
+    success: () => {
+      showTripDetails(trip);
+    }
+  });
+};
+
+//////////////////////////
+// sorting ///////////////
+//////////////////////////
+const render = function render(tripList) {
+  const tripListElement = $('#trip-list');
+  tripListElement.empty();
+
+  tripList.forEach((trip) => {
+    const generatedHTML = tripTemplate(trip.attributes);
+    tripListElement.append($(generatedHTML));
+
+  });
+  $('th.sort').removeClass('current-sort-field');
+  $(`th.sort.${ tripList.comparator }`).addClass('current-sort-field');
+  $('.trip').on('click', fetchTripDetails);
+};
+
+//////////////////////////
+// Add a Trip  ///////////
+//////////////////////////
+
+const readFormData = function readFormData() {
+  const tripData = { id: null };
+  TRIP_FIELDS.forEach((field) => {
+    const inputElement = $(`#add-trip-form input[name=${ field }]`);
+    const value = inputElement.val();
+    if (value != '') {
+      tripData[field] = value;
+    }
+    inputElement.val('');
+  });
+  return tripData;
+};
+
+const handleValidationFailures = function handleValidationFailures(errors) {
+  for (let field in errors) {
+    for (let problem of errors[field]) {
+      reportStatus('error', `${field}: ${problem}`);
+    }
+  }
+};
+const addTripHandler = function(event) {
+  event.preventDefault();
+  const trip = new Trip(readFormData());
+  clearStatus();
+  if (!trip.isValid()) {
+    handleValidationFailures(trip.validationError);
+    return;
+  }
+  tripList.add(trip);
+
+  trip.save({}, {
+    success: (model, response) => {
+      console.log('successfully saved trip!');
+      reportStatus('success', 'Successfully saved trip!');
+      tripList.add(model);
+      modal.hide();
+    },
+    error: (model, response) => {
+      console.log('failed to save trip.');
+      console.log(response);
+      tripList.remove(model); // triggers an update (and rerender) server-side validation failed
```

Comment:
Consider instead doing `model.destroy();`.  That will remove it from the list, and tell any listeners to stop.